lib/driver.js

```diff
@@ -1,407 +1,388 @@
+  constructor(context) {
+    super();
+    this.name = 'datastax';
+
+    this.consistencyLevel = {};
+    this.hostConfigKey = 'contactPoints';
+
+    this.types = cqlTypes.valueTypes;
+    this.dataType = _.extend(
+      {
+        objectAscii: -1,
+        objectText: -2
+      },
+      cqlTypes.dataTypes);
+
+    this.consistencyLevel = {
+      ONE: consistencies.one,
+      one: consistencies.one,
+      TWO: consistencies.two,
+      two: consistencies.two,
+      THREE: consistencies.three,
+      three: consistencies.three,
+      QUORUM: consistencies.quorum,
+      quorum: consistencies.quorum,
+      LOCAL_QUORUM: consistencies.localQuorum,
+      localQuorum: consistencies.localQuorum,
+      LOCAL_ONE: consistencies.localOne,
+      localOne: consistencies.localOne,
+      EACH_QUORUM: consistencies.eachQuorum,
+      eachQuorum: consistencies.eachQuorum,
+      ALL: consistencies.all,
+      all: consistencies.all,
+      ANY: consistencies.any,
+      any: consistencies.any
+    };
 
-DatastaxDriver.prototype.init = function init(context) {
-  if (!context) {
-    throw new Error('missing context object');
-  }
-  if (!context.config) {
-    throw new Error('missing context.config object');
+    this._init(context || {});
   }
 
-  /* istanbul ignore next: no benefit of testing coalesce */
-  this.config = context.config || {};
-  this.logger = context.logger || fakeLogger;
-  this.metrics = context.metrics;
-  if (context.config.queryDirectory) {
-    this.queryCache = new QueryCache({ queryDirectory: context.config.queryDirectory });
+  _init(context) {
+    if (!context.config) {
+      throw new Error('missing context.config object');
+    }
+
+    this.logger = context.logger || fakeLogger;
+    this.metrics = context.metrics;
+    this._initDriverConfig(context.config);
+
+    if (this.config.queryDirectory) {
+      this.queryCache = new QueryCache({ queryDirectory: this.config.queryDirectory });
+    }
+
+    this._initProviderOptions();
+    this.keyspace = this.config.keyspace;
+    this._initPoolConfig();
+    this._initConnectionResolver(context);
+
+    this.pools = {};
+    this.resultTransformers = context.resultTransformers || [];
   }
 
-  this.initProviderOptions(this.config);
-  this.keyspace = context.config.keyspace;
-  this.initPoolConfig();
-  this.initConnectionResolver(context);
+  _initDriverConfig(config) {
+    config.retryOptions = config.retryOptions || { retries: 0 };
+    const retryDelay = Math.max(config.retryDelay || 100, 0);
+    if (config.numRetries) {
+      config.retryOptions = {
+        retries: Math.max(config.numRetries, 0),
+        factor: 1,
+        minTimeout: retryDelay,
+        maxTimeout: retryDelay,
+        randomize: false
+      };
+    }
+    config.supportsPreparedStatements = true;
+    this.config = config;
+  }
 
-  this.pools = {};
-  this.resultTransformers = context.resultTransformers || [];
-};
+  _initProviderOptions() {
```

Comment:
This may be an anachronism from the dueling CQL drivers prior to the official driver being released. I believe the purpose was to translate the interface from helenus-style to node-cassandra-cql style. Since this is a major version bump, it might be a good time to get rid of this.

Reply:
@scommisso: I've removed a lot of the remapping logic now, besides the connection resolver part which has stayed the same. There is now a migration document that explains how to change your config to address the breakage.